src/Microsoft.Identity.Web.TokenAcquisition/TokenAcquisition.cs

```diff
@@ -599,8 +599,24 @@ private IConfidentialClientApplication BuildConfidentialClientApplication(Merged
 
             try
             {
+                static IList<string> AddContinuousAccessEvaluationCapability(IEnumerable<string> clientCapabilities)

```

Comment:
I don't think it should be done in TokenAcquisition. It should be done in DownstreamApi.

ITokenAcquirer and IAuthorizationHeaderProvider are not CAE capable if the app or library using them is not. DownstreamApi is CAE capable

Reply:
+1. DownstreamApi parses the WWW-authenticate header, and is CAE capable.

Reply:
What's the preferred way to pass the CAE flag from DownstreamApi.cs to TokenAcquisition?

Reply:
Also what about these two scenarios:
- User flow in a web app. If the CAE flag is moved to DownstreamApi level, when a user sign into the site non-CAE token gets cached. Then DownstreamApi.GetTokenForUser will get the cached token before calling the API.
- Singleton token acquisition. If the CCA gets created without the CAE flag, then we're stuck into that configuration.

Reply:
Optimal way forward is for MSAL.NET to expose `WithClientCapabilities` on each request. Then in Id Web, we can use it.

Reply:
cca requests, specifically

Reply:
We discussed a few options, @pmaytak will follow up with a detailed list: 

- `WithClientCapabilities` was always designed to be cache agnostic, MSAL has no idea if an access token is CAE-enabled or not. 
- We could add this functionality as an opt-in to MSAL, where MSAL differentiates between CAE capable / CAE non-capable tokens. It's not trivial and we'll take a small perf hit too.
- DownstreamAPI is ZTA but TokenAcquisition is not ZTA, so today Id.Web based website samples are not ZTA, but S2S samples are.
- Wondering if we can still make the website samples ZTA / CAE-capable by adding a "CAE capable" flag maybe in `AddMicrosoftIdentityWebApp` ?